gulpfile.js

```diff
@@ -68,6 +70,39 @@ gulp.task('watch', function () {
   }
 });
 
+// Watch Server files for changes
+gulp.task('watch_server_files_for_testing', function () {
+  var runOnlyChangedTestFile = argv.onlyChanged ? true : false;
+
+  // Start livereload
+  plugins.livereload.listen();
+
+  // Add Server Test file rules
+  gulp.watch([testAssets.tests.server, defaultAssets.server.allJS], ['test:server']).on('change', function (file) {
+    // check if we should only run a changed test file
+    if (runOnlyChangedTestFile) {
+      var changedTestFiles = [];
+
+      // iterate through server test glob patterns
+      _.forEach(testAssets.tests.server, function (tests) {
```

Comment:
@ilanbiala What do you think of this section overall?

Here, I'm iterating through the glob patterns because, since it's an array, technically a project could have multiple locations/patterns for tests.